### PR TITLE
Track more additional information

### PR DESCRIPTION
**Pull request by badlandspray**
Track `duration` as per the API [spec](https://listenbrainz.readthedocs.io/en/latest/users/json.html#payload-json-details). Store `spotify_id` in extra
Added `track_mbid` which was requested in #149 
Sorry for the minimal PR changes.

### CODE REVIEW

**Comment by badlandspray** [comment]
@krateng Should this be used as duration if duration is not present?

**Comment by krateng** [comment]
From the API docs, it looks like `duration` is the total length of the track, is that correct? It should be `track_length` then.

Is there any reason you need the other fields? `track_mbid` is a good idea, but I think all the mbids already define the track fully and I'm not sure we want to store proprietary details like Spotify's ID.

**Comment by badlandspray** [comment]
Yes, `duration` is the total length of the track in seconds.

The extension I use [sends](https://github.com/web-scrobbler/web-scrobbler/blob/master/src/core/background/scrobbler/listenbrainz-scrobbler.js#L271-L298) the above information in the payload as per the listenbrainz [API](https://listenbrainz.readthedocs.io/en/latest/users/json.html#payload-json-details) and seeing this as a replacement for the listenbrainz API, I thought it'd make sense to have these fields.

**Comment by krateng** [comment]
The API is of course meant to be compatible, so it should accept those fields, but Maloja itself doesn't need to store all information that Listenbrainz stores.

File: maloja/apis/listenbrainz.py [hunk 1]
```diff
@@ -77,6 +77,7 @@ def submit(self,pathnodes,keys):
 					'track_title':titlestr,
 					'album_name':albumstr,
 					'scrobble_time':timestamp,
+					'track_length': additional.get("duration"),
```
Comment (badlandspray) [inline maloja/apis/listenbrainz.py:80]: 1. Without `scrobble_duration` key, duration column is always NULL in `scrobbles` table. 
2. Even though `track_length` is present in `rawscrobble`, if you export the data from admin view track["length"] is null
Is this the correct behavior?
```
{
"track" : {
  "length": null
}
...
} 
```

Comment (krateng) [reply maloja/apis/listenbrainz.py:80]: The specification of Maloja is that scrobble_duration should only be saved if it's known, the assumption that scrobbles without that value have their track length as scrobble duration should only be made at the evaluation stage (when stats are generated).

This way the data itself isn't polluted by values that could be mixed between actual information and assumptions.

**Pull request merged**